github-pullrequest-plugin/src/main/java/org/jenkinsci/plugins/github/pullrequest/trigger/check/PullRequestToCauseConverter.java

```diff
@@ -42,12 +42,18 @@ public static PullRequestToCauseConverter toGitHubPRCause(GitHubPRRepository loc
         return new PullRequestToCauseConverter(localRepo, listener, trigger);
     }
 
+    /**
+     * @return only real trigger cause if matched trigger (not skip) event found for this remotePr.
+     */
     @Override
     public GitHubPRCause apply(final GHPullRequest remotePR) {
-        return from(trigger.getEvents())
+        final GitHubPRCause gitHubPRCause = from(trigger.getEvents())
                 .transform(toCause(remotePR))
                 .filter(notNull())
-                .firstMatch(new SkippedCauseFilter(listener)).orNull();
+                .first()
+                .get();
+
+        return gitHubPRCause.isSkip() ? null : gitHubPRCause;
```

Comment:
Won't the call to `get()` here throw an exception if no triggers match (when all events evaluate to a `null` cause)?

Also, this does have the side effect of evaluating all the events -- even after it finds one that matches. This might contradict the [documentation](https://github.com/jenkinsci/github-integration-plugin/blob/daa2b17f23433e21e5b5cb06fa5b95113d46d2ff/github-pullrequest-plugin/src/main/resources/org/jenkinsci/plugins/github/pullrequest/GitHubPRTrigger/help-events.html#L8-L10) which implies that events defined after a matching event are not evaluated.

Since `FluentIterable` doesn't have a `reduce` method, the only way I was able to get this working by those rules was to unwrap the fluent chain into a regular `for(...)` loop and break out at first match.

I updated PR #57 with that change... Feel free to take it or leave it.